include/drivers/pcie/pcie.h

```diff
@@ -81,18 +81,17 @@ extern bool pcie_probe(pcie_bdf_t bdf, pcie_id_t id);
  * @param bdf the PCI(e) endpoint
  * @param index (0-based) index
  * @param mbar Pointer to struct pcie_mbar
- * @return true if the mbar was found, false otherwise
+ * @return true if the mbar was found and is valid, false otherwise
  *
  * A PCI(e) endpoint has 0 or more memory-mapped regions. This function
- * allows the caller to enumerate them by calling with index=0..n. If
- * false is returned, there are no further regions. The indices
- * are order-preserving with respect to the endpoint BARs (skips useless bars
- * and I/O Bars): e.g., index 0 will return the lowest-numbered valid memory BAR
- * on the endpoint.
+ * allows the caller to enumerate them by calling with index=0..n.
+ * Value of n cannot be above 6, as there is a maximum of 6 BARs. The indices
```

Comment:
Value of n cannot be above 5, since the index is 0-based

Reply:
oh yes, it's badly written. I had in mind while (n < 6).